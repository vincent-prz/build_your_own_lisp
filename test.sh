#!/bin/bash
cc -std=c99 -Wall -g lispy.c mpc.c -ledit -lm -o lispy && ./lispy < test_input.txt > tmp.txt

# TODO: try to remove the first 3 lines of the comparison, with `tail -n +4 <file>
diff tmp.txt test_expected.txt
if [ $? -ne 0 ]
then
  echo "test failed"
  rm tmp.txt
  exit 1
else
  echo "test successful"
  rm tmp.txt
  exit 0
fi
