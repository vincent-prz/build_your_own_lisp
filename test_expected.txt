Lispy Version 0.0.0.0.2
Press Ctrl+c to Exit

()
4
39
-100
Error: Division By Zero!
Error: first element is not a function
{1 2 3 4}
{1 2 (+ 5 6) 4}
{{2 3 4} {1}}
Error: Function 'head' passed too many arguments. Got 2, Expected 1.
Error: Function 'head' passed incorrect types. Got Number, Expected Q-Expression.
Error: Function 'head' passed {}!
{1}
Error: Function 'tail' passed too many arguments. Got 2, Expected 1.
Error: Function 'tail' passed incorrect types. Got Number, Expected Q-Expression.
Error: Function 'tail' passed {}!
{2}
{1 2 3}
{1 2 {3}}
{}
{1 2}
{2}
{1 (+ 5 6)}
Error: Function 'join' passed incorrect type!
3
Error: Function 'eval' passed incorrect type!
Error: Function 'eval' passed too many arguments!
{head (list 1 2 3 4)}
{1}
{tail tail}
{6 7}
3
<builtin>
<builtin>
30
Error: Unbound Symbol hello
Error: Function 'def' should be supplied at least 2 arguments
Error: Function 'def' should be supplied a QExpr as a first argument
Error: Function 'def' should be supplied as much variable names as values to assign
Error: Function 'def' should be supplied as much variable names as values to assign
Error: Function 'def' should be supplied a QExpr with symbols as its children
()
100
()
1
2
()
{a b x y}
Error: Lambda passed incorrect number of arguments. Got 1, Expected 2.
Error: Lambda definition got incorrect argument type at position 0. Got Number, Expected Q-Expression.
Error: Lambda definition got incorrect argument type for formal argument 0. Got Number, Expected Symbol.
(\ {x y} {+ x y})
3
()
43
()
3
